website_compassion/forms/fund_donation_form.py

```diff
@@ -0,0 +1,106 @@
+##############################################################################
+#
+#    Copyright (C) 2021 Compassion CH (http://www.compassion.ch)
+#    @author: Jonathan Guerne <guernej@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import models, fields, _
+
+
+class FundDonationForm(models.AbstractModel):
+    _name = "cms.form.fund.donation"
+    _inherit = ["cms.form.payment"]
+
+    _display_type = "full"
+    _form_model = "res.partner"
+
+    amount = fields.Float(required=True)
+    fund = fields.Many2one("product.product")
+
+    _form_model_fields = [
+        "amount",
+    ]
+    _form_fields_hidden = ("invoice_id", "fund")
+
+    def _form_validate_amount(self, value, **kwargs):
+        if not value or not value.isnumeric() or float(value) < 1:
+            return "amount", _("Please set an amount")
+        return 0, 0
+
+    @property
+    def _payment_success_redirect(self):
+        return f"/my/new/donation/validate/{self.invoice_id.id}?payment=success"
+
+    @property
+    def _payment_error_redirect(self):
+        return f"/my/new/donation/validate/{self.invoice_id.id}?payment=error"
```

Comment:
Maybe add a TODO to check what we can do because I think this function is never called. There is a page with the list of cancelled invoice instead.